packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -643,60 +644,148 @@ export class SDKClient {
-        throw new SDKClientError({}, `${requestIdPrefix} Created file is empty. `);
+        if (callData.length > 0 && fileSize.isZero()) {
+          throw new SDKClientError({}, `${requestIdPrefix} Created file is empty. `);
+        }
+        this.logger.trace(`${requestIdPrefix} Created file with fileId: ${fileId} and file size ${fileSize}`);
       }
-      this.logger.trace(`${requestIdPrefix} Created file with fileId: ${fileId} and file size ${fileSize}`);
+
+      return fileId;
+    } catch (error: any) {
+      const sdkClientError = new SDKClientError(error, error.message);
+      let transactionFee: number | Hbar = 0;
+
+      // if valid network error utilize transaction id
+      if (sdkClientError.isValidNetworkError()) {
+        try {
+          const transactionCreateRecord = await new TransactionRecordQuery()
+            .setTransactionId(fileCreateTx.transactionId!)
+            .setNodeAccountIds(fileCreateTx.nodeAccountIds!)
+            .setValidateReceiptStatus(false)
+            .execute(this.clientMain);
+          transactionFee = transactionCreateRecord.transactionFee;
+          this.hbarLimiter.addExpense(transactionFee.toTinybars().toNumber(), currentDateNow);
+
+          this.captureMetrics(
+            SDKClient.transactionMode,
+            fileCreateTx.constructor.name,
+            sdkClientError.status,
+            transactionFee.toTinybars().toNumber(),
+            transactionCreateRecord?.contractFunctionResult?.gasUsed,
+            callerName,
+            interactingEntity,
+          );
+
+          if (fileAppendTx === null || fileAppendTx === undefined) {
+            this.logger.info(
+              `${requestIdPrefix} ${fileCreateTx.transactionId} ${callerName} ${fileCreateTx.constructor.name} status: ${sdkClientError.status} (${sdkClientError.status._code}), cost: ${transactionFee}`,
+            );
+          } else {
+            const transactionAppendRecord = await new TransactionRecordQuery()
+              .setTransactionId(fileAppendTx.transactionId!)
+              .setNodeAccountIds(fileAppendTx.nodeAccountIds!)
+              .setValidateReceiptStatus(false)
+              .execute(this.clientMain);
+            transactionFee = transactionAppendRecord.transactionFee;
+            this.hbarLimiter.addExpense(transactionFee.toTinybars().toNumber(), currentDateNow);
+
+            this.captureMetrics(
+              SDKClient.transactionMode,
+              fileCreateTx.constructor.name,
+              sdkClientError.status,
+              transactionFee.toTinybars().toNumber(),
+              transactionCreateRecord?.contractFunctionResult?.gasUsed,
+              callerName,
+              interactingEntity,
+            );
+
+            this.logger.info(
+              `${requestIdPrefix} ${fileCreateTx.transactionId} ${callerName} ${fileCreateTx.constructor.name} status: ${sdkClientError.status} (${sdkClientError.status._code}), cost: ${transactionFee}`,
```

Comment:
Based on above comment this would be updated as follows inside a if (fileAppendTx( {...}
```suggestion
              `${requestIdPrefix} ${fileAppendTx.transactionId} ${callerName} ${fileCreateTx.constructor.name} status: ${sdkClientError.status} (${sdkClientError.status._code}), cost: ${transactionFee}`,
```